docs/basics/primitives.mdx

```diff
@@ -95,6 +96,16 @@ The `updateValue` takes just one argument, which will be passed
 to the third argument of the write function of the atom.
 The behavior depends on how the write function is implemented.
 
+**Note:** as mentioned in the *atom* section, you have to take care of handling the reference of your atom, otherwise it may enter an infinite loop
+```typescript
+const safeAtom = atom(0)
+const Component = () => {
+  const [value] = useAtom(atom(0)) // Will cause an infinite loop
+  const [safeValue] = useAtom(safeAtmo) // is fine
+  const [alsoSafeValue] = useAtom(React.useMemo(0, [])) // is also fine
```

Comment:
You mean `useMemo(() => atom(0), [])` ?
This is fine, but useMemo doesn't have a semantic guarantee.
So, what we should show as a usage with useMemo is a derived atom.
```js
useMemo(() => atom((get) => get(countAtom) * 2), [])
```

Reply:
@dai-shi Yep forgot the inner function of useMemo :p Good point on semantic guarantee. Shouldn't we just show an example with useRef then ?

Okay your example has the merit of showing that it may make sense to create atoms on the fly. Myself never even thought of the idea. Nice !